v21.2/set-vars.md

```diff
@@ -4,35 +4,64 @@ summary: The SET statement modifies the current configuration variables for the
+<span class="version-tag">New in v21.2</span>: CockroachDB supports setting session variables for the duration of a single transaction, using [the `LOCAL` keyword](#set-local).
 
+{{site.data.alerts.callout_danger}}
+In some cases, client drivers can drop and restart the connection to the server. When this happens, any session configurations made with <code>SET</code> statements are lost. It is therefore more reliable to configure the session in the client's connection string. For examples in different languages, see the <a href="example-apps.html">Build an App with CockroachDB</a> tutorials.
+{{site.data.alerts.end}}
 
 ## Required privileges
 
-No [privileges](authorization.html#assign-privileges) are required to modify the session settings.
+No [privileges](authorization.html#assign-privileges) are required to modify the session variables.
 
 ## Synopsis
 
+The `SET` statement can set a session variable for the duration of the current session ([`SET (variable)`/`SET SESSION (variable)`](#set-session)), or for the duration of a single transaction ([`SET LOCAL (variable)`](#set-local)).
+
+{{site.data.alerts.callout_info}}
+The <code>SET</code> statement for session variables is unrelated to the other <a href="set-transaction.html"><code>SET TRANSACTION</code></a> and <a href="cluster-settings.html#change-a-cluster-setting"><code>SET CLUSTER SETTING</code></a> statements.
+{{site.data.alerts.end}}
+
+### SET SESSION
+
+<div>
+  {% include {{ page.version.version }}/sql/generated/diagrams/set_session.html %}
+</div>
+
+{{site.data.alerts.callout_info}}
+By default, session variables are set for the duration of the current session. As a result, [`SET (variable)` and `SET SESSION (variable)`](#set-session) are equivalent.
+{{site.data.alerts.end}}
+
+### SET LOCAL
+
+<span class="version-tag">New in v21.2</span>
+
 <div>
-  {% include {{ page.version.version }}/sql/generated/diagrams/set_var.html %}
+  {% include {{ page.version.version }}/sql/generated/diagrams/set_local.html %}
 </div>
 
-{{site.data.alerts.callout_info}}The <code>SET</code> statement for session settings is unrelated to the other <a href="set-transaction.html"><code>SET TRANSACTION</code></a> and <a href="cluster-settings.html#change-a-cluster-setting"><code>SET CLUSTER SETTING</code></a> statements.{{site.data.alerts.end}}
+{{site.data.alerts.callout_info}}
+`SET LOCAL` is compatible with [savepoints](savepoint.html).
 
-## Parameters
+Executing a [`ROLLBACK`](rollback-transaction.html) statement rolls back any variables set by `SET LOCAL`. Executing a `ROLLBACK TO SAVEPOINT` statement **does not** roll back variables set by `SET LOCAL` in the transaction.
```

Comment:
it does :o 

```
demo@127.0.0.1:26257/defaultdb> begin;
BEGIN


Time: 0ms total (execution 0ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> set time zone +3;
SET


Time: 1ms total (execution 0ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> savepoint s1;
SAVEPOINT


Time: 0ms total (execution 0ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> show time zone;
  timezone
------------
  3
(1 row)


Time: 1ms total (execution 1ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> set local time zone +2;
SET


Time: 1ms total (execution 0ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> show time zone;
  timezone
------------
  2
(1 row)


Time: 1ms total (execution 1ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> rollback to savepoint s1;
ROLLBACK


Time: 0ms total (execution 0ms / network 0ms)

demo@127.0.0.1:26257/defaultdb  OPEN> show time zone;
  timezone
------------
  3
(1 row)


Time: 1ms total (execution 1ms / network 0ms)

```

`RELEASE SAVEPOINT` also rolls `SET LOCAL` affected statements back to the savepoint